src/Nethermind/Nethermind.Network/Rlpx/RlpxHost.cs

```diff
@@ -213,10 +213,15 @@ private void InitializeChannel(IChannel channel, ISession session)
             }
             pipeline.AddLast("enc-handshake-handler", handshakeHandler);
 
-            channel.CloseCompletion.ContinueWith(x =>
+            channel.CloseCompletion.ContinueWith(async x =>
             {
+                // The close completion is completed before actual closing or remaining packet is processed.
+                // So usually, we do get a disconnect reason from peer, we just receive it after this. So w need to
+                // add some delay to account for whatever that is holding the network pipeline.
+                await Task.Delay(TimeSpan.FromSeconds(1));
```

Comment:
Does it have to be delay? Channel has a lot of own API's, maybe something there could be useful?

Reply:
Hmm, tried using a handler.  Does not work for some reason. May I'm missing something. I'll try again.